gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsRecord.java

```diff
@@ -176,14 +177,23 @@ public Builder name(String name) {
     }
 
     /**
-     * Sets the number of seconds that this record can be cached by resolvers. This number must be
-     * non-negative.
+     * Sets the time that this record can be cached by resolvers. This number must be non-negative.
+     * The maximum duration must be equivalent to at most {@link Integer#MAX_VALUE} seconds.
      *
-     * @param ttl A non-negative number of seconds
+     * @param duration A non-negative number of time units
+     * @param unit     The unit of the ttl parameter
      */
-    public Builder ttl(int ttl) {
-      checkArgument(ttl >= 0, "TTL cannot be negative. The supplied value was %s.", ttl);
-      this.ttl = ttl;
+    public Builder ttl(int duration, TimeUnit unit) {
+      checkArgument(duration >= 0,
+          "Duration cannot be negative. The supplied value was %s.", duration);
+      checkNotNull(unit);
+      // convert to seconds and check that we are not overflowing int
+      // we cannot do that because pb does not support it
+      long converted = unit.toSeconds(duration);
```

Comment:
Could be done by `Ints.checkedCast`(http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/primitives/Ints.html#checkedCast(long))